### ISSUE TITLE
Adjust PaintColrGlyph to directly express subset of layers

### ISSUE THREAD

**Issue by rsheeter** [googlefonts/nanoemoji#165]
In playing with PaintColrGlyph for Noto Emoji I realized that to reuse a subset of layers from a glyph you end up having to extract a new BaseGlyphV1Record and point all users at it. This makes reuse of a small # of layers result in an increase in filesize.

Visualization of what the layer graphs currently end up looking like @ https://codepen.io/rs42/pen/qBNRwew?editors=1000.

Thinking "aloud", what if we amend format 5 to directly express of "reuse some layers"? - sketch:

```C++
// 5 bytes. 
// In a LayerV1List each pointer costs 4 bytes
// For a single paint you just point directly, for 2+ consecutive you "win" by using format 5
PaintColrSlice
  uint8 format // 5
  uint16 glyphID
  uint8 firstLayer
  uint8 lastLayer
```

In this model the first clock face happily draws it's layers, the second comes along and wherever it has a run of at least 2 of the same layers it sticks in a format 5 entry.

**Comment by rsheeter** [comment]
Sanity checked locally and confirmed clock faces (heavy reuse of layers) do much better with the revised format 5.

**Comment by anthrotype** [comment]
I'm supportive of the idea.

> to reuse a subset of layers from a glyph you end up having to extract a new BaseGlyphV1Record

not only that, but since BaseGlyphV1List only contains unique records for each glyphID, when you want to reuse a subset of layers from a base color glyph you also have to allocate a new "virtual" glyphID at the font level (a scarce resource in 16-bit world), to be able to identify the layer subset and refer to it from PaintColorGlyph.

**Comment by anthrotype** [comment]
> much better

we should post here some numbers before and after this proposal (maybe also taking into account the impact of woff2 compression on this)

**Comment by rsheeter** [comment]
Filed https://github.com/googlefonts/colr-gradients-spec/issues/86 for the spec update, keeping this issue to track updating nanoemoji to take advantage of the revised spec.

**Comment by anthrotype** [comment]
https://github.com/googlefonts/colr-gradients-spec/issues/86 added new `PaintColrLayers` format (plus a global `LayerV1List`) that can be used to refer to slices of layers.
This is implemented in #164 

For `PaintColrGlyph` proper, we may want to add support for referencing a whole `PaintColrGlyph` when a color glyph reuses _all_ of the layers from another color glyph (not just a slice).

**Comment by rsheeter** [comment]
Spec proposal got amended for this, compiler is updated, I believe this is done.
